splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/TriggerHandler.java

```diff
@@ -147,37 +316,89 @@ public void firePendingAfterTriggers(Callable<Void> flushCallback) throws Except
          * Which is what we want. Check constraints before firing after triggers. */
         try {
             flushCallback.call();
+            if (getTriggerTempTableflushCallback() != null)
+                getTriggerTempTableflushCallback().call();
         } catch (Exception e) {
             pendingAfterRows.clear();
             throw e;
         }
 
-        for (ExecRow flushedRow : pendingAfterRows) {
-            fireAfterRowTriggers(flushedRow);
+        if (!hasAfterRow)
+            return;
+
+        List<Future<Void>> futures = new ArrayList<>();
+
+        // The LCC can't be shared amongst threads, so
+        // only use one level of concurrency for now.
+        if (true || pendingAfterRows.size() <= 1) {
```

Comment:
That's correct.  The else branch results in sharing of an LCC between threads which is not allowed.  I kept the code in as a reminder of a potential improvement, in case we find a way to enable the else branch in the future.  We could remove the else branch if we want to keep the code clean.